magento/magento-cloud-patches
# Update Symfony dependencies for compatibility with Magento 2.4 OnPrem

### Description
Currently is not possible to install magento-cloud-patches and magento/quality-patches on Magento 2.4 OnPrem instance due to Symfony composer dependencies conflicts.

Need to allow ^5.1 for "symfony/config", "symfony/dependency-injection", "symfony/proxy-manager-bridge".

### Fixed Issues (if relevant)

[MC-38123](https://jira.corp.magento.com/browse/MC-38123) Update MCP Symfony dependencies for compatibility with Magento 2.4 OnPrem

### Manual testing scenarios
Automated tests should pass.

It's possible to install composer package 'magento/magento-cloud-patches' on local Magento 2.4.0 instance:

2.4.0
1. composer create-project --repository-url=https://repo.magento.com/ magento/project-enterprise-edition=2.4.0
2. composer config repositories.mcp git git@github.com:magento-mpi/magento-cloud-patches.git
3. composer require "magento/magento-cloud-patches=dev-MC-38123 as 1.0.99"
4. Expected: magento/magento-cloud-patches composer package is installed

2.4.1
1. composer create-project --repository-url=https://repo.magento.com/ magento/project-enterprise-edition=2.4.1
2. composer config repositories.mcp git git@github.com:magento-mpi/magento-cloud-patches.git
3. composer require "magento/magento-cloud-patches=dev-MC-38123 as 1.0.99"
4. Expected: magento/magento-cloud-patches composer package is installed

### Release notes

None

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
